packages/email/src/render.ts

```diff
@@ -0,0 +1,34 @@
+import ejs from 'ejs';
+import * as fs from 'fs/promises';
+import mjmml2html from 'mjml';
+import winston from 'winston';
+
+import { Template } from './templates/template.type';
+import { Generic } from './templates/generic';
+import { welcomeMessage } from './templates/welcomeMessage';
+
+// Use default logger and set transporter instead of using custom logger.
+const console = new winston.transports.Console();
+winston.add(console);
+
+export async function render(template: Generic): Promise<string>;
+export async function render(template: welcomeMessage,): Promise<string>;
+export async function render(template: Template): Promise<string> {
+  const path = `src/templates/${template.name}/template.ejs`;
+
+  // this is probably lil weird no? @niekcandaele
```

Comment:
IMO it is better to just try and read the file. If it doesn't exist, it'll throw an error which can be handled. The way this is coded right now has race condition issues, right?